### PR TITLE
ChangesReducer Optimization

### PR DESCRIPTION
**Pull request by BenTauTau**
use `Set` contains rather than `Array` contains to reduce the time complexity

1. Before Optimize
image

2. After Optimize
image

### CODE REVIEW

File: Sources/ComposedUI/Common/ChangesReducer.swift [hunk 1]
```diff
@@ -375,13 +375,13 @@ internal struct ChangesReducer: CustomReflectable {
...
-        let itemsRemoved = changeset.elementsRemoved.filter(isIncluded(indexPath:)).map(\.item)
-        let itemsInserted = changeset.elementsInserted.filter(isIncluded(indexPath:)).map(\.item)
+        let itemsRemoved = changeset.elementsRemoved.filter(isIncluded(indexPath:))
+        let itemsInserted = changeset.elementsInserted.filter(isIncluded(indexPath:))
 
         let availableSpaces = (0..<Int.max)
             .lazy
-            .filter { !itemsRemoved.contains($0) }
-        let item = item - itemsInserted.filter({ $0 < item }).count
+            .filter { !itemsRemoved.contains(IndexPath(item: $0, section: section)) }
```
Comment (BenTauTau) [inline Sources/ComposedUI/Common/ChangesReducer.swift:383]: @JosephDuffy just want to check with you that the behavior will as expected after we use `IndexPath` to compare.
I think we have used `isIncluded` to filter out the same section so we can init a `IndexPath` here with section from function parameter.

Comment (JosephDuffy) [reply Sources/ComposedUI/Common/ChangesReducer.swift:383]: Yeah this looks to be the same and we have a lot of tests for this type so if they're passing then I trust it 👍

**Review by JosephDuffy — APPROVED** [review]
This looks like a good change and I can see how a very large array could cause some slowdowns.

I wonder if there are any other places that are converting to an array and we're seeing a performance hit 🤔

**Comment by BenTauTau** [comment]
> I wonder if there are any other places that are converting to an array and we're seeing a performance hit

I'm not quit sure about it now. However, I will continue to monitor the main project to see if there are any performance issues when using Composed. Based on different situations, we can decide how to optimize.

**Pull request merged**